### PR TITLE
Add Azure SQL profiler.

### PR DESCRIPTION
**Pull request by ykobuk**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,11 +16,22 @@ FROM python AS build
 COPY poetry.lock pyproject.toml ./
 
 # For pyodbc
+ENV ACCEPT_EULA=Y
+RUN apt-get update -y && apt-get install -y gnupg2
```
Comment (Vixtir) [inline Dockerfile]: I found RUN apt-get update 3 delete unnecessary.

File: Dockerfile [hunk 2]
```diff
@@ -16,11 +16,22 @@ FROM python AS build
 COPY poetry.lock pyproject.toml ./
```
Comment (Vixtir) [inline Dockerfile]: move that command to 34 line.

File: odd_collector_profiler/profilers/azure_sql/profiler.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+class AzureSQLProfiler(Profiler, SQLDialect):
+    """Profiler for Postgresql datasource"""
+
+    config_model = AzureSQLConfig
+    data_frame_reader = TableDataframeReader
+
+    def __init__(self, config: Dict[str, Any], data_profiler: DataProfiler):
+        super().__init__(config, data_profiler)
+        self.generator = AzureSQLGenerator(
+            host_settings=f"{self.config.server}.database.windows.net:{self.config.port}",
```
Comment (Vixtir) [inline odd_collector_profiler/profilers/azure_sql/profiler.py]: Is it necessary set `database.windows.net` ? Every server hosted here?

File: odd_collector_profiler/profilers/azure_sql/profiler.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            databases=self.config.database,
+        )
+
+    def get_statistics(self) -> DatasetStatisticsList:
+        items: List[DataSetStatistics] = []
+
+        with self.connect() as connection:
+            query = "SELECT name FROM sys.schemas WHERE principal_id=1"
```
Comment (Vixtir) [inline odd_collector_profiler/profilers/azure_sql/profiler.py:36]: Explicit what is principal_id=1

**Review by Vixtir — CHANGES_REQUESTED** [review]

File: odd_collector_profiler/profilers/azure_sql/profiler.py (hunk 3 above)
Comment (ykobuk) [reply odd_collector_profiler/profilers/azure_sql/profiler.py]: It's a part of the server name.
Example of full a connection string:
**Driver={ODBC Driver 18 for SQL Server};Server=tcp:my-test-server-ykobuk.database.windows.net,1433;Database=my_test_ykobuk;Uid=ykobuk_admin_server;Pwd={your_password_here};Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;**

File: Dockerfile (hunk 1 above)
Comment (ykobuk) [reply Dockerfile]: Done.

File: Dockerfile (hunk 2 above)
Comment (ykobuk) [reply Dockerfile]: Moved

File: odd_collector_profiler/profilers/azure_sql/profiler.py (hunk 4 above)
Comment (ykobuk) [reply odd_collector_profiler/profilers/azure_sql/profiler.py:36]: added comment:
The principal_id field is an ID of the principal that owns this schema.

**Review by Vixtir — APPROVED** [review]

**Pull request merged**